example/intkey_multiply/processor/Dockerfile-scar

```diff
@@ -0,0 +1,94 @@
+# Copyright 2019 Cargill Incorporated
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM ubuntu:bionic
+
+# Install base dependencies
+ENV VERSION=AUTO_STRICT
+
+RUN apt-get update \
+ && apt-get install -y \
+ curl \
+ gcc \
+ libssl-dev \
+ libzmq3-dev \
+ pkg-config \
+ unzip
+
+ENV PATH=$PATH:/protoc3/bin:/root/.cargo/bin
+
+# Install Rust
+RUN curl https://sh.rustup.rs -sSf > /usr/bin/rustup-init \
+ && chmod +x /usr/bin/rustup-init \
+ && rustup-init -y
+
+RUN rustup update \
+ && rustup target add wasm32-unknown-unknown
+
+# Install protoc
+RUN curl -OLsS https://github.com/google/protobuf/releases/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip \
+ && unzip -o protoc-3.7.1-linux-x86_64.zip -d /usr/local \
+ && rm protoc-3.7.1-linux-x86_64.zip
+
+# Create empty cargo project for sdk
+WORKDIR /build
+RUN USER=root cargo new --lib sdk
+
+# Copy over Cargo.toml, build.rs and protos
+COPY sdk/Cargo.toml /build/sdk/Cargo.toml
+COPY sdk/build.rs /build/sdk/build.rs
+COPY sdk/protos /build/sdk/protos
+
+# Do a release build to cache dependencies
+WORKDIR /build/sdk
+RUN cargo build --target wasm32-unknown-unknown --release
+
+WORKDIR /build
+
+# Create empty cargo project for intkey multiply
+RUN USER=root cargo new --bin example/intkey_multiply/processor
+
+# Copy over Cargo.toml file
+COPY example/intkey_multiply/processor/Cargo.toml \
+     /build/example/intkey_multiply/processor/Cargo.toml
+
+# Do a release build to cache dependencies
+WORKDIR /build/example/intkey_multiply/processor
+RUN cargo build --target wasm32-unknown-unknown --release
+
+# Remove the auto-generated .rs files and the built files
+RUN rm src/*.rs
+RUN rm target/wasm32-unknown-unknown/release/intkey* \
+    target/wasm32-unknown-unknown/release/deps/intkey*
+
+# Copy over source files
+COPY sdk/src /build/sdk/src
+COPY example/intkey_multiply/processor/src \
+     /build/example/intkey_multiply/processor/src
+
+# Build the contract
+RUN cargo build --target wasm32-unknown-unknown --release
+
+# Copy the packaging directory
+COPY example/intkey_multiply/processor/packaging/scar/* \
+     /build/example/intkey_multiply/processor/packaging/scar/
+
+# Copy the contract to the packaging directory
+RUN cp target/wasm32-unknown-unknown/release/intkey-multiply.wasm \
+    packaging/scar
+
+WORKDIR /build/example/intkey_multiply/processor/packaging/scar
+
+# Create .scar file
+RUN tar -jcvf /tmp/intkey-multiply_1.0.0.scar .
```

Comment:
This should be the version number of the smart contract in Cargo.toml - 0.4.3 currently.

Reply:
Should match in the .yaml manifest as well

Reply:
The version in the .yaml file is essentially the protocol version of the smart contract. So, it would only rarely match based on coincidence. Usually we use X.Y with no minor version for the smart contract protocol version, so really there is no chance they would match in practice.

Reply:
We might want to switch to a single-digit protocol version to make it clearer.

Reply:
I updated the file name.